release/authorization/README.md

```diff
@@ -0,0 +1,419 @@
+# gNMI Authorization Protocol
+
+## Objective
+
+This proto definition and the reference code(to be delivered seperately) serve
+to describe an authorization framework for controlling which OpenConfig paths of
+a network device users can access. The authorization policy is initially intended
+to be deployed to a device, with the ability to define:
+
+*   Policy rules - each rule defines a single authorization policy.
+*   Groups of users - as a method to logically group users in the administrative
+    domain, for instance: operators or administrators.
+*   Users which are individual users as part of either rules or in groups.
+
+Authentication information is not included in this Authorization configuration.
+
+Policy rules are matched based on the best match for the authorization request,
+not the first match against a policy rule. Best match enables a configuration
+which permits a user or group access to particular OpenConfig paths while
+denying subordinate portions of the permitted paths, or the converse, without
+regard to ordering of the rules in the configuration.
+
+## Best Match
+
+Authorization is performed for a singular user, gNMI path accessed, and access
+methodology (READ/WRITE). The result of an Authorization evaluation is an
```

Comment:
Is `READ`/`WRITE` sufficient here? I think that it could be, but wanted to understand how we expect that knits with the per-RPC security.  AFAICS, `READ` will correspond to both `Subscribe` and `Get`, and `WRITE` will correspond to `Set`. Is it intentional that there is no way to say `READ via Subscribe`?

Why might we care about this? `Get` on `/` might be extremely expensive for a device such that we want to prevent a user having `READ` access to `/` via Get, but `Subscribe` may be significantly cheaper and therefore allowed in the policy.  Do you envisage that there is a separate per-RPC policy somewhere that would allow that?

Reply:
We can add a SUBSCRIBE your example seems like a sane reason to do this. I think by the time the request gets to the authorization request you'd have to have brought along the method (GET(SUBSCRIBE)/SET) in order to tell if the authorization is for READ/WRITE.

Making the 'subscribe' explicit and not implicit in 'get' makes sense.
(added to proto)